### PR TITLE
New Roles to facilitate creation of FreeIPA sudoers group and rule 

### PR DESCRIPTION
**Pull request by jimright**
2 new roles in:
* **freeipa_host_group** to add the FreeIPA server to a inventory host group (currently just for AWS but with TODOs for GCP & Azure).
* **sudoers** for adding a specified list of users to a FreeIPA group and then creating a sudo rule for that group

### CODE REVIEW

File: roles/sudoers/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# global file for sudoers
+
+# Role prefix is 'sudoers__'
+
+# Variables used as inputs in main.yml
+sudoers__env_admin_username:                     "{{ cdp_workload_user }}"
+sudoers__env_admin_password:                     "{{ globals.admin_password | mandatory }}"
```
Comment (wmudge) [inline roles/sudoers/defaults/main.yml:23]: This is now a shared variable with https://github.com/cloudera-labs/cloudera.exe/blob/devel/roles/platform/defaults/main.yml#L66. Please pull this variable up into `cloudera.exe.common` and have both downstream roles map to this new upstream variable.

File: roles/sudoers/meta/main.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+galaxy_info:
+  author: Jim Enright (jenright@cloudera.com)
+  description: >
+    Adding specified users to FreeIPA sudoers group and creation of a sudo rule to allow passwordless sudo for the group. Existing users in the sudoers can be purged or retained depending on the value of the sudoers__purge_users_in_group flag.
```
Comment (wmudge) [inline roles/sudoers/meta/main.yml:18]: Format this line to within 120 chars.

**Review by wmudge — CHANGES_REQUESTED** [review]

File: roles/sudoers/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# global file for sudoers
+
+# Role prefix is 'sudoers__'
+
+# Variables used as inputs in main.yml
+sudoers__env_admin_username:                     "{{ cdp_workload_user }}"
```
[the remark this answers is not in the document]
Comment (jimright) [reply roles/sudoers/defaults/main.yml]: Should be fixed in https://github.com/cloudera-labs/cloudera.exe/pull/6/commits/40d424a3a31f6f3bf9b6d2314bc811de5c5b2a8c

File: roles/sudoers/defaults/main.yml (hunk 1 above)
Comment (jimright) [reply roles/sudoers/defaults/main.yml]: Should be fixed in https://github.com/cloudera-labs/cloudera.exe/pull/6/commits/40d424a3a31f6f3bf9b6d2314bc811de5c5b2a8c

File: roles/sudoers/meta/main.yml (hunk 2 above)
Comment (jimright) [reply roles/sudoers/meta/main.yml]: Should be fixed in https://github.com/cloudera-labs/cloudera.exe/pull/6/commits/40d424a3a31f6f3bf9b6d2314bc811de5c5b2a8c

File: roles/sudoers/tasks/teardown.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Remove the FreeIPA sudo rule
+- name: Remove sudo rule for passwordless sudo
+  community.general.ipa_sudorule:
+    ipa_user: "{{ sudoers__env_admin_username }}"
```
Comment (wmudge) [inline roles/sudoers/tasks/teardown.yml:20]: `sudoers__env_admin_username` is not defined in `teardown.yml`. I suggest you pull out the username discovery into a shared task list like `initialize.yml` and invoke in both `main.yml` and `teardown.yml`.

File: roles/sudoers/tasks/teardown.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    ipa_pass: "{{ sudoers__env_admin_password }}"
+    name: "{{ sudoers__sudorule_name }}"
+    state: absent
+
+# Remove the FreeIPA group for sudo
+- name: Remove sudoers group
+  community.general.ipa_group:
+    ipa_user: "{{ sudoers__env_admin_username }}"
```
Comment (wmudge) [inline roles/sudoers/tasks/teardown.yml:28]: See above

**Review by wmudge — CHANGES_REQUESTED** [review]

File: roles/sudoers/tasks/teardown.yml (hunk 5 above)
Comment (jimright) [reply roles/sudoers/tasks/teardown.yml:32]: Fixed as part of https://github.com/cloudera-labs/cloudera.exe/pull/6/commits/488871c7128bd4a4f1b98b180929a4901d7851e0

File: roles/freeipa_host_group/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  ansible.builtin.set_fact:
+    __cdp_workload_username: "{{ __cdp_iam_current_user_info.users[0].workloadUsername }}"
+
+# Add the FreeIPA server and username to the inventory
+- name: Add FreeIPA servers to inventory
+  ansible.builtin.add_host:
+    name: "{{ item }}" 
+    groups: "{{ freeipa_host_group__host_group_name }}" 
+    ansible_user: "{{ __freeipa_server_login_user }}"
+    cdp_workload_user: "{{ __cdp_workload_username }}"
```
Comment (wmudge) [inline roles/freeipa_host_group/tasks/main.yml:82]: Is this still needed? Actually, do we need to add anything to the host entry other than the `name` and `groups`?

**Review by wmudge — CHANGES_REQUESTED** [review]

File: roles/freeipa_host_group/tasks/main.yml (hunk 6 above)
Comment (jimright) [reply roles/freeipa_host_group/tasks/main.yml]: Removed `ansible_user` and `cdp_workload_user` variables as part of commits https://github.com/cloudera-labs/cloudera.exe/pull/6/commits/26cf7fa7b40481ac1eeb1769407c31d67906a530 and https://github.com/cloudera-labs/cloudera.exe/pull/6/commits/377b2a4389eda9d98540691b83bcdd0ccf9c6b9e

**Pull request merged**